fixes #11568 - adds help scripts to recovery methods

cfouant opened the pull request.

cfouant committed:
fixes #11568 - adds help scripts to recovery methods

komidore64 commented on katello/katello-backup:
```diff
@@ -1,94 +1,113 @@
-  echo "Usage: katello-backup <backup directory>"
-  echo " eg: $ katello-backup /tmp/katello-backup/"
+  echo "Usage: katello-backup --directory <backup directory>"
+  echo " eg: $ katello-backup --directory /tmp/katello-backup/"
```
why are you changing this from being a positional argument? specifying the directory is required, correct?

cfouant replied:
I felt like it was nicer in case down the line we wanted to add other arguments to the script. Open for discussion however.

komidore64 replied:
Other arguments are fine. Then wouldn't they just follow before or after the directory argument?

cfouant replied:
They would follow after the directory argument. It's doable, this is easier for me and opens it up for user, but I can change it do katello-backup <directory> --option1 --option2 if you feel that's nicer.

komidore64 replied:
What does "opens it up for the user", mean?

komidore64 replied:
The point that I am attempting (poorly) to get across is that arguments are for required information that a CLI tool needs, and options are, well, optional.

I realize that Hammer uses options for both required and non-required information completely throwing out the entire notion of positional arguments which drives me bonkers, but that ship has already sailed and I doubt anyone would hear me out to change that now.

If you're wanting to use `--directory DIR` instead of requiring a positional `/path/to/dir` argument for the sake of using the same style as Hammer, then I can live with that.

cfouant replied:
Freedom to add arguments in whatever fashion they would like... 
katello-backup --option1 --directory --option2

cfouant replied:
I'm happy to go to a positional argument, I had asked for feedback on argument vs option and didn't really hear anything back so just went with options.

komidore64 replied:
Okay, well if you asked and I didn't chime in. Options are fine.

cfouant replied:
Too late I'm already updating it :)

komidore64 commented on katello/katello-backup:
```diff
@@ -1,94 +1,113 @@
-echo ""
+  # Start services
+  service postgresql restart
+  katello-service restart
```
Since we know that these services are being stopped above on lines 41 and 42 maybe we should call `start` on these instead of `restart`?

komidore64 commented on katello/katello-restore:
```diff
@@ -1,77 +1,89 @@
+  # Restart services
+  echo "Restarting all Red Hat Satellite processes"
+  service postgresql restart
+  katello-service restart
```
`start` vs `restart`?

stbenjam commented on katello/katello-restore:
```diff
@@ -1,77 +1,89 @@
-echo "Done."
-echo ""
+  # Restart services
+  echo "Restarting all Red Hat Satellite processes"
```
Katello\* (or remove product names)

komidore64 replied:
:+1:

cfouant commented:
@stbenjam @komidore64 changes made: altered script to ruby, removed flag for directory (making it an argument in the command rather than a flag), removed references to product names. Let me know what you guys think!

stbenjam commented:
My feedback looks addressed :smile:  I'll leave it to @komidore64 to comment on the rest

komidore64 commented on katello/katello-backup:
```diff
@@ -1,94 +1,88 @@
-cd $BDIR
-echo "Creating backup folder $BDIR"
-echo ""
+  opts.on_tail("-h", "--help", "Show this help message and exit") do
```
`OptionParser` handles help output for you, you shouldn't need to specify it here.

komidore64 replied:
i was thinking of clamp, not optionparser. my mistake.

cfouant replied:
From what I have seen, you need to define all options, including help.

cfouant replied:
I stand corrected!

komidore64 commented:
Were you going to parse flags and options at first and then decided not to? It seems like you're using OptionParser only for help output, which isn't wrong. It formats help output quite nicely with little effort from us.

cfouant commented:
I will be adding options shortly, but wanted to address specific but regarding --help flags first.

komidore64 commented on katello/katello-backup:
```diff
@@ -1,94 +1,83 @@
+else
+  FileUtils.mkdir_p(DIR)
+  puts "Creating backup folder #{DIR}"
+  `chgrp postgres #{DIR}`
```
since we're already using `FileUtils` for other things, any objection to changing this to `FileUtils.chown(nil, 'postgres', DIR)`?

komidore64 commented:
looks good to me

ack

komidore64 commented:
@ehelms does this need any magic for SCL shenanigans?

ehelms commented:
Is it intended to run under the SCL? i.e. can these scripts run under 1.8.7?

komidore64 commented:
all the rest of our ruby is under the scl, correct? would/should this be an exception?

ehelms commented:
Not all -- katello-service script for example isn't.

cfouant commented:
hmmm yeah i hadn't thought of that, do i need to change the katello.spec file?

komidore64 commented:
@cfouant i think it's fine like it is.

ehelms commented:
If it runs on EL6, as is, then we are all good.

komidore64 commented:
@cfouant you can confirm?

cfouant commented:
let me do another run through since changes made

cfouant commented:
I had to change one line in the restore scripts from Dir.exist? to File.directory? as the exist method does not exist for the version of ruby in EL6.

EDIT: @ehelms @komidore64 @stbenjam - not sure if I need final ACKs from you guys, given the code change (listed here)

cfouant commented:
@ehelms @komidore64 @stbenjam do i need final ACKs from you? I did have to make a change to get everything working on EL6

stbenjam commented:
My feedback was addressed so ACK from me
